apps/site/src/components/pages/wordpress/early-access-cta.tsx

```diff
@@ -0,0 +1,161 @@
+import { faCheck } from "@fortawesome/free-solid-svg-icons";
+import { inputBaseClasses, Theme, useMediaQuery } from "@mui/material";
+// Our custom TextField hides the 'endAdornment' when the 'error' prop is true
+// eslint-disable-next-line no-restricted-imports
+import TextField from "@mui/material/TextField";
+import { MouseEvent, TouchEvent, useMemo, useRef, useState } from "react";
+
+import { apiClient } from "../../../lib/api-client";
+import { Button } from "../../button";
+import { useEmailTextField } from "../../hooks/use-email-text-field";
+import { ArrowRightIcon, FontAwesomeIcon } from "../../icons";
+import { useEmailSubmitted } from "./email-submitted-context";
+
+const submittedEmailText = "You’re already on the waitlist as";
+const submitErrorText = "There was an error submitting your email";
+
+export const EarlyAccessCTA = () => {
+  const { submittedEmail, setSubmittedEmail } = useEmailSubmitted();
+
+  const emailInputRef = useRef<HTMLInputElement>(null);
+
+  const isSmall = useMediaQuery(({ breakpoints }: Theme) =>
+    breakpoints.down("sm"),
+  );
+
+  const {
+    emailValue,
+    setEmailValue,
+    isEmailValid: isEmailInputValid,
+    emailHelperText,
+  } = useEmailTextField({});
+
+  const [loading, setLoading] = useState<boolean>(false);
+  const [submitted, setSubmitted] = useState<boolean>(false);
```

Comment:
Do you need this state? Couldn't you just use `submittedEmail` to check if the user has submitted?

Reply:
Probably not, unless we don't want to show the `You’re already on the waitlist as {email}` label on the form that was submitted. What do you think?